src/shims/unix/fd.rs

```diff
@@ -564,7 +563,13 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         drop(fd);
 
         let result = result?.map(|c| i64::try_from(c).unwrap());
-        Ok(Scalar::from_target_isize(this.try_unwrap_io_result(result)?, this))
+        match result {
+            Ok(written_bytes) => Ok(Scalar::from_target_isize(written_bytes, this)),
+            Err(e) => {
+                this.set_last_error_from_io_error(e)?;
+                Ok(Scalar::from_target_isize(-1, this))
```

Comment:
This code got worse than it used to be, what's happening here?

Reply:
These are writing target sized integers, so i'll need to add a helper type to wrap the `i64`s in. The read method already could not be ported before and was written out fully. I want to fix both at the same time